### PR TITLE
[Owners] Adding meta data for APIs using ViAddr[]

### PR DESCRIPTION
**Pull request by kruthi-NI**
### Justification

Adding meta-data and generated files for the DCPower APIs which use ViInt32[]:

Task: https://ni.visualstudio.com/DevCentral/_workitems/edit/1417332

### Implementation

- In this PR, I am adding meta-data and generated files for 2 DCPower APIs (`ExportAttributeConfigurationBuffer`, `ImportAttributeConfigurationBuffer`) which use `ViAddr[]`.
- Adding meta-data was causing compile-time errors in the auto-generated `nidcpower_service.cpp` file.
- The error is similar to what is explained in PR #2 
- `ViAddr` is defined [as `void *` in `IviVisaType.h`](https://github.com/ni/grpc-device/blob/main/imports/include/IviVisaType.h#L118). But the grpc type used for `ViAddr` [is `fixed64` which is defined in proto_helpers.py](https://github.com/ni/grpc-device/blob/main/source/codegen/proto_helpers.py#L47) .
- Hence direct conversion between `ViAddr`  (which is internally void *) and `fixed64`(which is internally `google::protobuf::uint64` ) is not possible.
- This mismatch was causing a compile-time error for using `const_cast` in the auto-generated implementation of `ImportAttributeConfigurationBuffer` API.
- The error message said something like `const_cast cannot convert from const (google::protobuf::uint64)* to ViAddr*` . 
- Also note that `const_cast` can only adjust qualifiers by removing const keyword but cannot change the underlying data type. 

Note:  Before my changes, service.cpp.mako helper file was [using `const_cast` for all types of arrays](https://github.com/ni/grpc-device/blob/main/source/codegen/templates/service_helpers.mako#L190) as shown in below image:

const_cast

- To resolve this error, I modified the service.cpp.mako helper file to use `reinterpret_cast` before `const_cast` for `ViAddr[]` similar to what was done in PR #2.
- This mismatch was also causing compile-time error for the direct conversion in the auto-generated implementation of `ExportAttributeConfigurationBuffer` API. 
- To resolve that second error, I am using `reinterpret_cast` in the mako file location which was causing that second error.

### Testing

I created a client example code in Python which uses `ImportAttributeConfigurationBuffer` and `ExportAttributeConfigurationBuffer` APIs. I ran the example using a simulated DCPower device and verified that the APIs are working fine.

### CODE REVIEW

**Review by doshirohan — APPROVED** [review]

**Review by ganeshnie — APPROVED** [review]

File: source/codegen/templates/service_helpers.mako [hunk 1]
```diff
@@ -215,8 +215,11 @@ one_of_case_prefix = f'{namespace_prefix}{function_name}Request::{PascalFieldNam
       std::vector<${underlying_param_type}> ${parameter_name}(${size}, ${underlying_param_type}());
 %     elif service_helpers.is_string_arg(parameter):
       std::string ${parameter_name}(${size}, '\0');
-%     else:
+%     elif underlying_param_type == 'ViAddr':
       response->mutable_${parameter_name}()->Resize(${size}, 0);
+      ${underlying_param_type}* ${parameter_name} = reinterpret_cast<${underlying_param_type}*>(response->mutable_${parameter_name}()->mutable_data());
+%     else:
+      response->mutable_${parameter_name}()->Resize(${size}, 0); 
       ${underlying_param_type}* ${parameter_name} = response->mutable_${parameter_name}()->mutable_data();
```
Comment (kruthi-NI) [inline source/codegen/templates/service_helpers.mako:223]: Note: Alternative option is to add if else condition inside this else block which might look something like this:
alternative_1

But indentation did not look good to me in that approach.
 But if owners ask to change this, I can change

File: source/codegen/templates/service_helpers.mako [hunk 2]
```diff
@@ -187,7 +187,7 @@ one_of_case_prefix = f'{namespace_prefix}{function_name}Request::{PascalFieldNam
 % elif c_type == 'ViSession':
       auto ${parameter_name}_grpc_session = ${request_snippet};
       ${c_type} ${parameter_name} = session_repository_->access_session(${parameter_name}_grpc_session.id(), ${parameter_name}_grpc_session.name());\
-% elif c_type == 'ViInt32[]':
+% elif c_type == 'ViInt32[]' or c_type == 'ViAddr[]':
       auto ${parameter_name} = const_cast<${c_type_pointer}>(reinterpret_cast<const ${c_type_pointer}>(${request_snippet}.data()));\
```
Comment (jonlugoatx) [inline source/codegen/templates/service_helpers.mako:191]: I didn't notice this before. There's no point in doing a `const_cast` of a `reinterpret_cast`. We should just make it a `reinterpret_cast` without the `const`.

**Review by jonlugoatx — APPROVED** [review]

File: source/codegen/templates/service_helpers.mako (hunk 2 above)
Comment (kruthi-NI) [reply source/codegen/templates/service_helpers.mako:191]: @jonlugoatx ,
>  We should just make it a reinterpret_cast without the const.

We initially tried this way. But we got the following error saying that "reinterpret_cast cannot cast away const or other type qualifiers".

Comment_const

Then @ganeshnie suggested using const_cast before reinterpret_cast.

Comment (kruthi-NI) [reply source/codegen/templates/service_helpers.mako:191]: @jonlugoatx 
I discussed with @ganeshnie and we decided to resolve this comment and submit that other PR for now. If we need to make any changes, we can do it as part of this PR.

**Review by reckenro — APPROVED** [review]

**Pull request merged**